The AC_LANG sections upset libtool 2.2.2, easiest fix is to remove them.
Using a .cpp file in a Makefile means configure.ac must use AC_PROG_CXX
so patch out the section from the makefile too.

RP - 14/4/08

Index: gst-ffmpeg-0.10.1/gst-libs/ext/ffmpeg/configure.ac
===================================================================
--- gst-ffmpeg-0.10.1.orig/gst-libs/ext/ffmpeg/configure.ac	2008-04-14 23:55:25.000000000 +0100
+++ gst-ffmpeg-0.10.1/gst-libs/ext/ffmpeg/configure.ac	2008-04-14 23:56:18.000000000 +0100
@@ -742,26 +742,7 @@
 
 dnl Beos Audio
 AC_FF_ALLOW_DISABLE(CONFIG_AUDIO_BEOS, audio-beos, BeOS audio support,[
-  if test x$TARGET_OS = xbeos; then
-    AC_LANG_CPLUSPLUS
-    AC_CHECK_HEADER(SoundPlayer.h,[
-      OLD_CXXFLAGS="$CXXFLAGS"
-      CXXFLAGS="$CXXFLAGS -lbe -lmedia"
-      AC_MSG_CHECKING(For BeOS audio libraries)
-      AC_TRY_COMPILE([
-#include <Application.h>
-#include <SoundPlayer.h>
-        ],[
-          BSoundPlayer *p = new BSoundPlayer(NULL,NULL,NULL);
-        ],[ CONFIG_AUDIO_BEOS=yes && AC_MSG_RESULT(yes) ],[
-          CONFIG_AUDIO_BEOS=no && AC_MSG_RESULT(no)
-      ])
-      CXXFLAGS="$OLD_CXXFLAGS"
-    ], CONFIG_AUDIO_BEOS=no)
-    AC_LANG_C
-  else
     CONFIG_AUDIO_BEOS=no
-  fi
 ])
 
 dnl Check for freetype2, used in vook/libdrawtext.so
Index: gst-ffmpeg-0.10.1/gst-libs/ext/ffmpeg/libavformat/Makefile.am
===================================================================
--- gst-ffmpeg-0.10.1.orig/gst-libs/ext/ffmpeg/libavformat/Makefile.am	2008-04-14 23:58:44.000000000 +0100
+++ gst-ffmpeg-0.10.1/gst-libs/ext/ffmpeg/libavformat/Makefile.am	2008-04-14 23:58:55.000000000 +0100
@@ -24,13 +24,6 @@
 audiooss_SRC = audio.c
 endif
 
-if CONFIG_AUDIO_BEOS
-audiobeos_SRC = beosaudio.cpp
-audiobeos_LIBS = \
-	-lbe \
-	-lmedia
-endif
-
 if CONFIG_VIDEO4LINUX
 v4l_SRC = grab.c v4l2.c
 endif
